src/services/usercentrics-service.ts

```diff
@@ -27,36 +30,61 @@ const subscribe = (x: (x: ServiceState) => void): (() => void) => {
     }
 }
 
-const setIsOpen = (isOpen: boolean) => {
+const setIsOpen = (isOpen: boolean, hasInteracted?: boolean) => {
     state = { ...state, isOpen }
+    if (hasInteracted !== undefined) state.hasInteracted = hasInteracted
     update()
 }
-const setIsInitialized = (initialized: boolean) => {
-    state = { ...state, initialized }
+
+const setIsInitialized = (isInitialized: boolean) => {
+    state = { ...state, isInitialized }
     update()
 }
 
 const onEvent = (event: Event) => {
     if (isUCUICMPEvent(event)) {
         switch (event.detail.type) {
-            case UCUICMPEventType.CMP_SHOWN:
+            case UCUICMPEventType.CMP_SHOWN: {
```

Comment:
why add `{`

Reply:
No reason, I think at some point I was setting variables and wanted to make sure it didn't leak into the outer scope. In any case, removed 👍